### PR TITLE
Add missing custom fields for datacenter

### PR DESCRIPTION
**Pull request by ButternCream**
# Pull Request

To help us get this pull request reviewed and merged quickly, please be sure to include the following items:

* [x] Tests (if applicable)
* [x] Documentation (if applicable)
* [ ] Changelog entry
* [x] A full explanation here in the PR description of the work done

## PR Type
What kind of change does this PR introduce?

* [x] Bugfix
* [x] Feature
* [ ] Code style update (formatting, local variables)
* [ ] Refactoring (no functional changes, no api changes)
* [ ] Build related changes
* [ ] CI related changes
* [ ] Documentation content changes
* [ ] Tests
* [ ] Other

## Backward Compatibility

Is this change backward compatible with the most recently released version? Does it introduce changes which might change the user experience in any way? Does it alter the Connector in any way?

* [x] Yes (backward compatible)
* [ ] No (breaking changes)

#### Useful links:

- [https://github.com/ConductorOne/baton-sdk/wiki/Coding-Guidelines](Baton SDK coding guidelines)
- [https://github.com/ConductorOne/baton/blob/main/CONTRIBUTING.md](New contributor guide)

## What's new?
- Adds new createmeta endpoints from go-jira
- Pulls in required custom fields and maps them to our schemas

## Summary by CodeRabbit

- **New Features**
  - Enhanced error handling in issue creation and new methods for retrieving issue types and metadata fields.
  - Added support for mapping custom field IDs to Jira types.
  - Improved management of custom fields during ticket creation and retrieval, including conversion and filtering methods.

- **Bug Fixes**
  - Updated dependencies to the latest versions, which may include important bug fixes and performance improvements.

### CODE REVIEW

File: pkg/connector/tickets.go [hunk 1]
```diff
@@ -21,6 +22,168 @@ import (
...
+	type JiraPickerStruct struct {
+		Id string `json:"id"`
+	}
+
+	pickObjects := []*JiraPickerStruct{}
+
+	switch v := field.GetValue().(type) {
+	case *v2.TicketCustomField_StringValue:
+		strValue := v.StringValue.GetValue()
+		if d.cfIdToJiraType[field.GetId()] == jira.TypeUser {
```
Comment (btipling) [inline pkg/connector/tickets.go:40]: is this always safe?, should it be `if v, ok := d.cfIdToJiraType[field.GetId()]; ok {...`?

also below

File: pkg/connector/tickets.go [hunk 2]
```diff
@@ -21,6 +22,168 @@ import (
...
+		return v.StringValues.GetValues(), nil
+
+	case *v2.TicketCustomField_BoolValue:
+		return v.BoolValue.GetValue(), nil
+
+	case *v2.TicketCustomField_TimestampValue:
+		// must be in ISO 8601 date time format (RFC3339)
+		// https://developer.atlassian.com/server/jira/platform/jira-rest-api-example-create-issue-7897248/#datetimefield
+		// -> Date time picker custom field
+		return v.TimestampValue.GetValue().AsTime().Format(time.RFC3339), nil
```
Comment (btipling) [inline pkg/connector/tickets.go:67]: can `GetValue()` return nil?

**Review by btipling — APPROVED** [review]

File: pkg/connector/tickets.go (hunk 2 above)
Comment (ButternCream) [reply pkg/connector/tickets.go]: https://github.com/ConductorOne/baton-jira-datacenter/pull/28/commits/901e4d7306cf01130eec0d06a15154b51aaec709

File: pkg/connector/tickets.go (hunk 1 above)
Comment (ButternCream) [reply pkg/connector/tickets.go]: https://github.com/ConductorOne/baton-jira-datacenter/pull/28/commits/901e4d7306cf01130eec0d06a15154b51aaec709

**Pull request merged**